flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BlockableInput.java

```diff
@@ -26,9 +26,12 @@
 
 /**
  * Input, with just basic methods for blocking and resuming consumption. It can be for example an {@link InputGate}
+ * or a chained source.
  */
 @Internal
 public interface BlockableInput {
+	void blockConsumption(int inputChannelIdx);
+
 	void resumeConsumption(int channelIndex) throws IOException;
```

Comment:
`resumeConsumption` could be implicitly triggered by `checkpointStopped`.

Then, the last unfitting method is `blockConsumption`, which could be modeled as `void barrierReceived(int channelIndex, CheckpointBarrier barrier);` (and it wouldn't be odd that it's ignored by InputGate).

Reply:
As discussed offline, we decided to keep it as it is for now.